src/3d/qgs3dmapscene.cpp

```diff
@@ -472,6 +472,7 @@ void Qgs3DMapScene::createTerrain()
   if ( mTerrain )
   {
     mSceneEntities.removeOne( mTerrain );
+    mTerrain->stopAllJobs();
 
     mTerrain->deleteLater();
```

Comment:
what if we just replaced `mTerrain->deleteLater();` by `delete mTerrain;`, without adding stopAllJobs() ? The destructor is expected to cancel all pending jobs. If that would work, that would be a much simpler solution than adding stopAllJobs() and having `mIsValid` as a new state that chunked entity implementation needs to keep in mind...

Reply:
cool - if that works, please feel free to remove any extra code we don't need (addition of stopAllJobs()) and we can merge it...

Reply:
Done.